### ISSUE TITLE
Scheduler deploys pods on nodes with fully used cpu resources

### ISSUE THREAD

**Issue by robertbotez** [kubernetes/kubernetes#109405]
### What happened?

We tried to test the default functionality of the scheduler in order to implement a custom one. From our understanding, the scheduler should deploy the pods on the most available node. 

### What did you expect to happen?

We used 3 worker nodes in our cluster, and we stressed two of them using stress package from Linux and we expected that the pod should be deployed on the non-stressed node (we impose also the cpu request in the deployment manifest file). We saw that the pod was deployed on one of the two stressed nodes instead of the one with the most available resources.

### How can we reproduce it (as minimally and precisely as possible)?

Deploy a cluster with 3 worker nodes, with 4vcpu and 4 GB of RAM. Then stress two workers with stress tool from Linux and requests 3000m cpu for the deployment (we used a deployment for nginx) and see where the pods are being deployed.

### Kubernetes version

Client Version: version.Info{Major:"1", Minor:"23", GitVersion:"v1.23.5", GitCommit:"c285e781331a3785a7f436042c65c5641ce8a9e9", GitTreeState:"clean", BuildDate:"2022-03-17T03:51:43Z", GoVersion:"go1.17.8", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"22", GitVersion:"v1.22.3", GitCommit:"c92036820499fedefec0f847e2054d824aea6cd1", GitTreeState:"clean", BuildDate:"2021-10-27T18:35:25Z", GoVersion:"go1.16.9", Compiler:"gc", Platform:"linux/amd64"}

### Cloud provider

Bare-metal

### OS version

<details>

```console
# On Linux:
$ cat /etc/os-release
NAME="Ubuntu"
VERSION="20.04.4 LTS (Focal Fossa)"
ID=ubuntu
ID_LIKE=debian
PRETTY_NAME="Ubuntu 20.04.4 LTS"
VERSION_ID="20.04"
HOME_URL="https://www.ubuntu.com/"
SUPPORT_URL="https://help.ubuntu.com/"
BUG_REPORT_URL="https://bugs.launchpad.net/ubuntu/"
PRIVACY_POLICY_URL="https://www.ubuntu.com/legal/terms-and-policies/privacy-policy"
VERSION_CODENAME=focal
UBUNTU_CODENAME=focal
$ uname -a
Linux ubuntu 5.13.0-39-generic #44~20.04.1-Ubuntu SMP Thu Mar 24 16:43:35 UTC 2022 x86_64 x86_64 x86_64 GNU/Linux


</details>


### Install tools

kubeadm

### Container runtime (CRI) and version (if applicable)

Docker

### Related plugins (CNI, CSI, ...) and versions (if applicable)

Calico CNI

**Comment by yxxhero** [comment]
When Kube scheduler schedules pod, the actual load of nodes will not be considered,  it only considers the remaining resources and the relationship with cpu and mem `resource.request`.

**Comment by ahg-g** [comment]
Correct, the scheduler doesn't take into account the actual utilization of the node, only "requested" resources.

There are some external scheduler plugins that aim to provide such utilization-aware semantics; see this for example https://github.com/kubernetes-sigs/scheduler-plugins/tree/master/pkg/trimaran

**Comment by wajika** [comment]
I think this is a very good proposal, hopefully someone can add this feature and provide a switch control to turn it on if I need to.
